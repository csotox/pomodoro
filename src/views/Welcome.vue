<template>
  <section class="welcome">
    <div class="application__content welcome__content">
      <h1 class="welcome__title">
        Hola, ¡bienvenido a POMODORO!
      </h1>
      <h2 class="welcome__subtitle">
        Mejora tu productividad centrándote en lo importante de cada momento y sé más eficiente
      </h2>

      <Button @click="onClick()">
        Haz tu primer pomodoro
      </Button>
    </div>
  </section>
</template>

<script>
import Button from '@/components/Button.vue';

export default {
  name: 'Home',
  components: {
    Button,
  },
  methods: {
    onClick() {
      this.$router.push({ name: 'pomodoro', params: { type: 'work' } });
    },
  },
};
</script>

<style lang="scss" scoped>
.welcome {
  background-color: var(--application-color-primary);
}

.welcome__content {
  display: flex;
  flex-direction: column;
  justify-content: center;
  color: var(--application-color-text);
}

.welcome__title,
.welcome__subtitle {
  text-align: left;
}

.welcome__title {
  margin-bottom: 1.2em;
  font-size: 2.5rem;
  font-weight: 500;
  letter-spacing: 1px;
}

.welcome__subtitle {
  margin-bottom: 1.2em;
  font-size: 2rem;
  font-weight: 300;
}
</style>
