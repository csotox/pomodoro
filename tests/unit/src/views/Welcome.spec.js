import { shallowMount } from '@vue/test-utils';
import Welcome from '@/views/Welcome.vue';

describe('Welcome.vue', () => {
  it('should render properly', () => {
    const welcome = shallowMount(Welcome);
    expect(welcome).toMatchSnapshot();
  });

  describe('Methods', () => {
    it('should redirect to pomodoro route with work parameter', () => {
      const routerPushSpy = jest.fn();

      const welcome = shallowMount(Welcome, {
        mocks: {
          $router: {
            push: routerPushSpy,
          },
        },
      });

      welcome.vm.onClick();

      expect(routerPushSpy).toHaveBeenCalledWith({
        name: 'pomodoro',
        params: {
          type: 'work',
        },
      });
    });
  });
});
