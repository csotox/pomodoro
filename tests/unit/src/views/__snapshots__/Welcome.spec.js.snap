// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome.vue should render properly 1`] = `
<section class="welcome">
  <div class="application__content welcome__content">
    <h1 class="welcome__title">
      Hola, ¡bienvenido a POMODORO!
    </h1>
    <h2 class="welcome__subtitle">
      Mejora tu productividad centrándote en lo importante de cada momento y sé más eficiente
    </h2>
    <button-stub>
      Haz tu primer pomodoro
    </button-stub>
  </div>
</section>
`;
